pkg/agent/endpoints/workload/handler.go

```diff
@@ -178,6 +180,22 @@ func (h *Handler) ValidateJWTSVID(ctx context.Context, req *workload.ValidateJWT
 	}
 	log.WithField(telemetry.SPIFFEID, spiffeID).Debug("Successfully validated JWT")
 
+	id, err := spiffeid.FromString(spiffeID)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "unexpected SPIFFE ID: %v", err)
+	}
+
+	if !id.MemberOf(h.c.TrustDomain) {
+		filteredClaims := make(map[string]interface{})
+		for claim, value := range claims {
+			if isClaimAllowed(claim, h.c.AllowedForeignJWTClaims) {
+				filteredClaims[claim] = value
+			}
+		}
+
+		claims = filteredClaims
```

Comment:
Can we just prune the existing claims map instead of creating a new one?
```suggestion
		for claim := range claims {
			if !isClaimAllowed(claim, h.c.AllowedForeignJWTClaims) {
				delete(claims, claim)
			}
		}
```